COLLADAMaya/src/COLLADAMayaGeometryExporter.cpp

```diff
@@ -405,6 +408,226 @@ namespace COLLADAMaya
+            : Element(streamWriter, typeName)
+        {
+            streamWriter.appendValues(value);
+        }
+
+        Type(COLLADASW::StreamWriter& streamWriter, const String& typeName, const T& value, uint numValues)
+            : Element(streamWriter, typeName)
+        {
+            for (uint i = 0; i < numValues; ++i) {
+                streamWriter.appendValues(value[i]);
+            }
+        }
+    };
+
+    template<typename T>
+    class NewParam : public Element
+    {
+    public:
+        NewParam(COLLADASW::StreamWriter& streamWriter, const String& sid, const String& typeName, const T & value)
+            : Element(streamWriter, COLLADASW::CSWC::CSW_ELEMENT_NEWPARAM, sid)
+        {
+            Type<T> type(streamWriter, typeName, value);
+        }
+
+        NewParam(COLLADASW::StreamWriter& streamWriter, const String& sid, const String& typeName, const T & value, uint numValues)
+            : Element(streamWriter, COLLADASW::CSWC::CSW_ELEMENT_NEWPARAM, sid)
+        {
+            Type<T> type(streamWriter, typeName, value, numValues);
+        }
+    };
+
+    class ExtraAttributeExporter : public AttributeParser
+    {
+    public:
+        ExtraAttributeExporter(COLLADASW::StreamWriter& sw)
+            : mSW(sw)
+            , mAttributeLevel(0)
+        {}
+
+    protected:
+        int mAttributeLevel;
+
+        // AttributeParser overrides
+        
+        virtual bool onBeforeAttribute(MFnDependencyNode & fnNode, MObject & attr) override
+        {
+            ++mAttributeLevel;
+
+            MStatus status;
+            MFnAttribute fnAttr(attr, &status);
+            if (!status) return false;
+
+            MString attrName = fnAttr.name(&status);
+            if (!status) return false;
+
+            bool isDynamic = fnAttr.isDynamic(&status);
+            if (!status) return false;
+
+            if (!isDynamic)
+                return false;
+
+            bool isHidden = fnAttr.isHidden(&status);
+            if (!status) return false;
+
+            if (isHidden)
+                return false;
+
+            if (mAttributeLevel > 1) {
+                // Don't export compound attribute internal attributes as standalone newparam.
+                // Compound internal attributes are exported in onCompoundAttribute().
+                return false;
+            }
+
+            return true;
+        }
+
+        virtual void onAfterAttribute(MFnDependencyNode& node, MObject& attr) override
+        {
+            --mAttributeLevel;
+        }
+
+        virtual void onBoolean(MPlug & plug, const MString & name, bool value) override
+        {
+            NewParam<bool> e(mSW, name.asChar(), COLLADASW::CSWC::CSW_VALUE_TYPE_BOOL, value);
+        }
+
+        virtual void onByte(MPlug & plug, const MString & name, char value) override
+        {
+            char text[5];
+            sprintf(text, "0x%X", value);
```

Comment:
could you use snprintf to make sure output string is never more than 5 characters?